### PR TITLE
Add utility for generating random 64-bit numbers

### PR DESCRIPTION
**Pull request by rnburn**
The implementation uses thread-local random number generators and sets up a fork handler so that we can continue to generate unique IDs when the process is forked.

### CODE REVIEW

File: sdk/test/common/random_fork_test.cc [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#ifdef __unix__
+// Verifies that IDs don't clash after forking the process.
+//
+// See https://github.com/opentracing-contrib/nginx-opentracing/issues/52
+#  include "src/common/random.h"
```
Comment (reyang) [inline sdk/test/common/random_fork_test.cc]: Curious - any reason for this weird spacing between `#` and `include`?

Comment (rnburn) [reply sdk/test/common/random_fork_test.cc]: It's from clang-format. But, I think, because it's nested in a #if

File: sdk/src/common/random.cc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  TlsRandomNumberGenerator() noexcept
+  {
+    Seed();
+    platform::AtFork(nullptr, nullptr, OnFork);
+  }
+
+  static std::mt19937_64 &engine() noexcept { return engine_; }
+
+private:
+  static thread_local std::mt19937_64 engine_;
```
Comment (g-easy) [inline sdk/src/common/random.cc]: This costs 2.5KB per thread.

File: sdk/src/common/platform/fork_windows.cc [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{
+  (void)prepare;
+  (void)parent;
+  (void)child;
+  return 0;
+}
+}  // namespace platform
+}  // namespace common
+}  // namespace sdk
+OPENTELEMETRY_BEGIN_NAMESPACE
```
Comment (g-easy) [inline sdk/src/common/platform/fork_windows.cc]: END :)

C:\Users\circleci\project\sdk\src\common\platform\fork_windows.cc(20): error C1075:  '{': no matching token found

File: sdk/src/common/random.cc (hunk 2 above)
Comment (rnburn) [reply sdk/src/common/random.cc]: I'd fine switching to a version like this
https://github.com/lightstep/lightstep-tracer-cpp/blob/master/src/common/fast_random_number_generator.h
It's smaller and faster. And for the use-case of generating unique IDs, I think, perfectly fine.

File: sdk/src/common/platform/fork_windows.cc (hunk 3 above)
Comment (rnburn) [reply sdk/src/common/platform/fork_windows.cc]: Ah, that's what I missed. Fixed. Thanks.

Comment (rnburn) [reply sdk/src/common/platform/fork_windows.cc]: I switched to FastRandomNumberGenerator.

It takes up only 16 bytes and here's the benchmark comparison
https://2145-193156399-gh.circle-artifacts.com/0/benchmark/sdk/test/common/random_benchmark_result.txt
```
2020-03-27 02:18:54
Running bazel-out/host/bin/sdk/test/common/random_benchmark
Run on (36 X 3370.14 MHz CPU s)
CPU Caches:
  L1 Data 32K (x18)
  L1 Instruction 32K (x18)
  L2 Unified 1024K (x18)
  L3 Unified 25344K (x1)
Load Average: 6.47, 9.48, 13.46
-------------------------------------------------------------------
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_RandomIdGeneration          2.13 ns         2.13 ns     63421653
BM_RandomIdStdGeneration       6.18 ns         6.18 ns     22676378
```
The algorithm [xorshift](https://en.wikipedia.org/wiki/Xorshift) is fine for ID generation.

File: sdk/src/common/random.cc (hunk 2 above)
Comment (rnburn) [reply sdk/src/common/random.cc]: I switched to FastRandomNumberGenerator.

It takes up only 16 bytes and here's the benchmark comparison
https://2145-193156399-gh.circle-artifacts.com/0/benchmark/sdk/test/common/random_benchmark_result.txt
```
2020-03-27 02:18:54
Running bazel-out/host/bin/sdk/test/common/random_benchmark
Run on (36 X 3370.14 MHz CPU s)
CPU Caches:
  L1 Data 32K (x18)
  L1 Instruction 32K (x18)
  L2 Unified 1024K (x18)
  L3 Unified 25344K (x1)
Load Average: 6.47, 9.48, 13.46
-------------------------------------------------------------------
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_RandomIdGeneration          2.13 ns         2.13 ns     63421653
BM_RandomIdStdGeneration       6.18 ns         6.18 ns     22676378
```
The algorithm [xorshift](https://en.wikipedia.org/wiki/Xorshift) is fine for ID generation.

Comment (g-easy) [reply sdk/src/common/random.cc]: Nice. Thanks for adding a benchmark also.

**Pull request merged**